cerbero/build/cookbook.py

```diff
@@ -426,7 +440,7 @@ def _reset_recipe_status(self, recipe, change_name, previous, now):
     def _reset_recipe_if_needed(self, recipe):
         # Check that the recipe still exists in cache, because
         # a prior recipe might have reset its status
-        if not recipe.name in self.status:
+        if not recipe.name in self.status or self._recipe_is_reset(recipe.name):
```

Comment:
checking if recipe.name is in self.status should be part of the is_reset function